sklearn/mixture/_base.py

```diff
@@ -48,6 +48,12 @@ class BaseMixture(DensityMixin, BaseEstimator, metaclass=ABCMeta):
     provides basic common methods for mixture models.
     """
 
+    # _init_skipped is intended to be used by
+    # derived classes to skip the initialization of parameters if all
+    # parameters are provided by the user. This is useful for example in
+    # BayesianGaussianMixture where the user can provide an initial value
+    _init_skipped = False
```

Comment:
I'm not a fan of adding more state if we can avoid it. In this case, I think we can get away with overriding `_initialize_parameters` as follows:

```patch
diff --git a/sklearn/mixture/_gaussian_mixture.py b/sklearn/mixture/_gaussian_mixture.py
index e0b630f37c..e8a8f88ff0 100644
--- a/sklearn/mixture/_gaussian_mixture.py
+++ b/sklearn/mixture/_gaussian_mixture.py
@@ -701,6 +701,17 @@ class GaussianMixture(BaseMixture):
                 n_features,
             )
 
+    def _initialize_parameters(self, X, random_state):
+        compute_resp = (
+            self.weights_init is None
+            or self.means_init is None
+            or self.precisions_init is None
+        )
+        if compute_resp:
+            super()._initialize_parameters(X, random_state)
+        else:
+            self._initialize(X, None)
+
     def _initialize(self, X, resp):
         """Initialization of the Gaussian mixture parameters.
 
@@ -712,10 +723,12 @@ class GaussianMixture(BaseMixture):
         """
         n_samples, _ = X.shape
 
-        weights, means, covariances = _estimate_gaussian_parameters(
-            X, resp, self.reg_covar, self.covariance_type
-        )
-        weights /= n_samples
+        if resp is not None:
+            weights, means, covariances = _estimate_gaussian_parameters(
+                X, resp, self.reg_covar, self.covariance_type
+            )
+            if self.weights_init is None:
+                weights /= n_samples
 
         self.weights_ = weights if self.weights_init is None else self.weights_init
         self.means_ = means if self.means_init is None else self.means_init

```

Reply:
Mr Fan. Thank you for your suggestion! I will rewrite this